src/nifs/sangria/accumulator.rs

```diff
@@ -11,20 +11,87 @@ use itertools::Itertools;
 use rayon::prelude::*;
 use tracing::{debug, instrument, warn};
 
-use super::{GetConsistencyMarkers, GetStepCircuitInstances};
+use super::{GetConsistencyMarkers, GetStepCircuitInstances, CONSISTENCY_MARKERS_COUNT};
 use crate::{
     commitment::CommitmentKey,
     ff::{Field, PrimeField},
-    ivc::instances_accumulator_computation,
+    ivc::sangria::instances_accumulator_computation,
+    main_gate::{AssignedValue, WrapValue},
     plonk::{
         self, GetChallenges, GetWitness, PlonkInstance, PlonkStructure, PlonkTrace, PlonkWitness,
     },
     poseidon::{AbsorbInRO, ROTrait},
     util::ScalarToBase,
 };
 
+/// Accumulator for step-circuit instance columns
+///
+/// Since the first column is folded directly via `r` and the rest (step-circuit instances columns)
```

Comment:
is the first column the marker column and the rest are user-defined  instance columns?